src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1676,11 +1693,40 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                     throw ExceptionUtilities.UnexpectedValue(node.OperatorResultKind);
             }
 
-            bool? resultIsNullable = getIsNullable(leftOperand, leftResult) == false ? false : getIsNullable(rightOperand, rightResult);
-            _resultType = TypeSymbolWithAnnotations.Create(resultType, resultIsNullable);
+            NullableAnnotation resultNullableAnnotation;
+
+            if (getValueNullableAnnotation(leftOperand, leftResult).IsAnyNotNullable())
```

Comment:
>getValueNullableAnnotation(leftOperand, leftResult) [](start = 16, length = 51)

I don't understand why we use `getValueNulalbleAnnotation(left)` for the `if` but then set the result type to `getNullableAnnotation(left)`. Please consider adding a comment for which cases that distinction is important.

Reply:
> I don't understand why we use getValueNulalbleAnnotation(left) for the if but then set the result type to getNullableAnnotation(left). Please consider adding a comment for which cases that distinction is important. 

We want to check if the value can be nullable based on annotations and getValueNullableAnnotation might adjust value that it returns to emphasize the fact. However, we want the original annotation to flow through the system. I will add a comment.

---
In reply to: [232115987](https://github.com/dotnet/roslyn/pull/30913#discussion_r232115987) [](ancestors = 232115987)